xcm/xcm-builder/src/location_conversion.rs

```diff
@@ -35,22 +35,26 @@ impl<Network: Get<NetworkId>, AccountId: From<[u8; 32]> + Into<[u8; 32]> + Clone
 	}
 }
 
-/// A [`MultiLocation`] consisting of a single `Parent` [`Junction`] will be converted to the
-/// default value of `AccountId` (e.g. all zeros for `AccountId32`).
-pub struct ParentIsDefault<AccountId>(PhantomData<AccountId>);
-impl<AccountId: Default + Eq + Clone> Convert<MultiLocation, AccountId>
-	for ParentIsDefault<AccountId>
+/// A [`MultiLocation`] consisting of a single `Parent` [`Junction`] will be converted to all zeros
+/// for `AccountId`.
+pub struct ParentIsAllZeroes<AccountId>(PhantomData<AccountId>);
+impl<AccountId: Decode + Eq + Clone> Convert<MultiLocation, AccountId>
+	for ParentIsAllZeroes<AccountId>
 {
 	fn convert_ref(location: impl Borrow<MultiLocation>) -> Result<AccountId, ()> {
 		if location.borrow().contains_parents_only(1) {
-			Ok(AccountId::default())
+			Ok(AccountId::decode(&mut TrailingZeroInput::zeroes())
+				.expect("infinite length input; no invalid inputs for type; qed"))
```

Comment:
For what is this account being used?

Reply:
It looks like it is only used in testing, in cases where the test wants to convert a message originating from a parent location to just the zero account, i.e. a "don't care but signed" account.

As such, there's not really a huge issue leaving it as `ParentIsDefault`, other than to keep in line with the `Default` bound removal on `AccountId`.

Reply:
I think we should remove this trait from the public api and make it test only.

Reply:
I think default account is always zero account.

Reply:
Yeah `Default == Zero`.

I'm also for removing this or doing it like @xlc proposed.